administration/src/bp-modules/self-service/hooks/useCardGeneratorSelfService.tsx

```diff
@@ -1,34 +1,22 @@
-import { ApolloError } from '@apollo/client'
-import React, { useCallback, useContext, useState } from 'react'
+import { useCallback, useContext, useState } from 'react'
 import { useTranslation } from 'react-i18next'
 import { useSearchParams } from 'react-router-dom'
 
-import { Card, generateCardInfo, initializeCardFromCSV } from '../../../cards/Card'
+import { Card, initializeCardFromCSV } from '../../../cards/Card'
 import { generatePdf } from '../../../cards/PdfFactory'
-import { CreateCardsError, CreateCardsResult } from '../../../cards/createCards'
-import getMessageFromApolloError from '../../../errors/getMessageFromApolloError'
-import { DynamicActivationCode, StaticVerificationCode } from '../../../generated/card_pb'
+import { CreateCardsResult, createSelfServiceCard } from '../../../cards/createCards'
 import { useCreateCardsFromSelfServiceMutation } from '../../../generated/graphql'
 import { ProjectConfigContext } from '../../../project-configs/ProjectConfigContext'
 import { getCsvHeaders } from '../../../project-configs/helper'
-import { base64ToUint8Array, uint8ArrayToBase64 } from '../../../util/base64'
 import downloadDataUri from '../../../util/downloadDataUri'
-import getCustomDeepLinkFromQrCode from '../../../util/getCustomDeepLinkFromQrCode'
-import { reportErrorToSentry } from '../../../util/sentry'
 import { useAppToaster } from '../../AppToaster'
-import FormErrorMessage from '../components/FormErrorMessage'
+import { showCardGenerationError } from '../../util/cardGenerationError'
 
-export enum CardSelfServiceStep {
-  form,
-  information,
-  activation,
-}
+export type SelfServiceCardGenerationStep = 'input' | 'loading' | 'information' | 'activation'
```

Comment:
I faced the following issue with the enum and could not figure out how to solve this except changing it to a string type: https://github.com/webpack/webpack/issues/16146